packages/next/src/build/collect-build-traces.ts

```diff
@@ -623,6 +628,9 @@ export async function collectBuildTraces({
     }
 
     const { entryNameFilesMap } = buildTraceContext?.chunksTrace || {}
+    const infos =
+      pageInfos instanceof Map ? pageInfos : deserializePageInfos(pageInfos)
+
```

Comment:
The fact that this is potentially a worker boundary means the interface should be as small as possible. PageInfos is only used for one check so it’d be even better if we only passed the info needed. I’d just turn this into an array of edge runtime routes. We can construct this parallel data structure when we build page infos. The. I’ve thing is that if you aren’t using edge at all then we pay zero when crossing the worker boundary and when you do we only send the bare minimum info